### PR TITLE
DefaultHttp2RemoteFlowController reentry infinite loop

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
DefaultHttp2RemoteFlowController.writePendingBytes does not support reentry but does not enforce this constraint. Reentry is possible if the channel transitions from Writable -> Not Writable -> Writable during the distribution phase. This can happen if the user flushes when notified of the channel transitioning to the not writable state, and may be done if the user wants to fill the channel outbound buffer before flushing.

Modifications:
- DefaultHttp2RemoteFlowController.writePendingBytes should protect against reentry

Result:
DefaultHttp2RemoteFlowController will not allocate unexpected amounts or enter an infinite loop.

### CODE REVIEW

**Comment by Scottmitch** [comment]
@nmittler - PTAL

File: codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.http2.StreamByteDistributor.Writer;
 import io.netty.util.BooleanSupplier;
+import io.netty.util.internal.PlatformDependent;
```
Comment (netkins) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java]: MINOR Remove this unused import 'io.netty.util.internal.PlatformDependent'. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by normanmaurer** [comment]
LGTM after addressing netkins commen

**Comment by nmittler** [comment]
@Scottmitch LGTM

**Comment by Scottmitch** [comment]
4.1 (9de90d0)

**Pull request closed without merging**